CRM/MembershipExtras/Job/OfflineAutoRenewal.php

```diff
@@ -163,16 +165,31 @@ public function run() {
         $this->currentInstallmentsNumber = $recurContribution['installments'];
       }
 
-      $this->setLastContribution();
+      $tx = new CRM_Core_Transaction();
 
-      if ($this->currentInstallmentsNumber > 1) {
-        $this->renewWithInstallmentsPaymentPlan();
-      }
-      else {
-        $this->renewNoInstallmentsPaymentPlan();
-      }
+      try {
+        $this->setLastContribution();
+
+        if ($this->currentInstallmentsNumber > 1) {
+          $this->renewWithInstallmentsPaymentPlan();
+        }
+        else {
+          $this->renewNoInstallmentsPaymentPlan();
+        }
+
+        $this->dispatchMembershipRenewalHook();
+      } catch (Exception $e) {
+        $tx->rollback();
+        $message = "An error ocurred renewing a payment plan with id({$recurContribution['contribution_recur_id']}): " . $e->getMessage();
+
+        CRM_Core_Session::setStatus(
```

Comment:
I don't see the value of this, the scheduled job will usually run in the background as a cron job so there is no need to set the session status since it does not make sense in the context of the command line, but maybe you can store the error in civicrm log file.